src/main/java/org/jabref/gui/desktop/JabRefDesktop.java

```diff
@@ -168,7 +169,38 @@ private static void openExternalFilePlatformIndependent(Optional<ExternalFileTyp
      * @throws IOException
      */
     public static void openFolderAndSelectFile(Path fileLink) throws IOException {
-        NATIVE_DESKTOP.openFolderAndSelectFile(fileLink);
+        if (Objects.isNull(fileLink)) {
+            return;
+        }
+        boolean usingDefault = Globals.prefs.getBoolean(JabRefPreferences.USE_DEFAULT_FILE_BROWSER_APPLICATION);
+
+        if(usingDefault) {
+            NATIVE_DESKTOP.openFolderAndSelectFile(fileLink);
+        }else{
```

Comment:
There are certain code style errors, which is resulting in failed tests.

Reply:
Yes you right, I didn't included checkstyle in my IDE, I'll fix all the checkstyle issues in next commit.